### PR TITLE
[Keymap] Add pierrec83's gherkin keymap

### PR DESCRIPTION
**Pull request by pierrechevalier83**
Contribute my gherkin keymap.

## Description

Contribute my gherkin keymap upstream as it is semi-stable. It has grown
in symbiosis with my Kyria keymap which is already upstream.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/40percentclub/gherkin/keymaps/pierrec83/readme.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+* Load keymap.json into qmk configurator
+* Perform any edits
+* Export the keymap. This should save a json file in `~/Downloads/pierrec83.json` or equivalent for your OS
+* From the root of qmk_firmware, move the keymap to its destination:
+```
+mv ~/Downloads/pierrec83.json keyboards/40percentclub/gherkin/keymaps/pierrec83/keymap.json
+```
+* Regenerate the `keymap.c`:
+```
+qmk json2c keyboards/40percentclub/gherkin/keymaps/pierrec83/keymap.json -o keyboards/40percentclub/gherkin/keymaps/pierrec83/keymap.c
```
Comment (zvecr) [inline keyboards/40percentclub/gherkin/keymaps/pierrec83/readme.md:33]:
```suggestion
```

File: keyboards/40percentclub/gherkin/keymaps/pierrec83/keymap.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#include QMK_KEYBOARD_H
```
Comment (zvecr) [inline keyboards/40percentclub/gherkin/keymaps/pierrec83/keymap.c:1]: This file should be deleted, the build process will automatically regenerate it.

**Comment by Erovia** [comment]
You can use your Configurator-compatible json as `keymap.json` without any manual conversion.

See: https://github.com/qmk/qmk_firmware/tree/master/keyboards/clueboard/california/keymaps/default

File: keyboards/40percentclub/gherkin/keymaps/pierrec83/keymap.c (hunk 2 above)
Comment (pierrechevalier83) [reply keyboards/40percentclub/gherkin/keymaps/pierrec83/keymap.c]: Oh! That is a great piece of information :)
Like you said, `qmk flash` still works without this extra step of generating the keymap manually. That's really cool!

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**